xilem_core/src/views/one_of.rs

```diff
@@ -110,36 +135,438 @@ pub trait OneOfCtx<
+                (OneOf9::G(v), OneOf9::G(ref mut state)) => {
+                    Context::with_downcast_g(&mut element, |element| {
+                        v.teardown(state, ctx, element);
+                    });
+                }
+                (OneOf9::H(v), OneOf9::H(ref mut state)) => {
+                    Context::with_downcast_h(&mut element, |element| {
+                        v.teardown(state, ctx, element);
+                    });
+                }
+                (OneOf9::I(v), OneOf9::I(ref mut state)) => {
+                    Context::with_downcast_i(&mut element, |element| {
+                        v.teardown(state, ctx, element);
+                    });
+                }
+                _ => unreachable!(),
+            }
+        });
+    }
+
+    #[doc(hidden)]
+    fn message(
+        &self,
+        view_state: &mut Self::ViewState,
+        id_path: &[ViewId],
+        message: Message,
+        app_state: &mut State,
+    ) -> MessageResult<Action, Message> {
+        let (start, rest) = id_path
+            .split_first()
+            .expect("Id path has elements for OneOf");
+        if start.routing_id() != view_state.generation {
+            return MessageResult::Stale(message);
+        }
+        match (self, &mut view_state.inner_state) {
+            (OneOf9::A(v), OneOf9::A(ref mut state)) => v.message(state, rest, message, app_state),
+            (OneOf9::B(v), OneOf9::B(ref mut state)) => v.message(state, rest, message, app_state),
+            (OneOf9::C(v), OneOf9::C(ref mut state)) => v.message(state, rest, message, app_state),
+            (OneOf9::D(v), OneOf9::D(ref mut state)) => v.message(state, rest, message, app_state),
+            (OneOf9::E(v), OneOf9::E(ref mut state)) => v.message(state, rest, message, app_state),
+            (OneOf9::F(v), OneOf9::F(ref mut state)) => v.message(state, rest, message, app_state),
+            (OneOf9::G(v), OneOf9::G(ref mut state)) => v.message(state, rest, message, app_state),
+            (OneOf9::H(v), OneOf9::H(ref mut state)) => v.message(state, rest, message, app_state),
+            (OneOf9::I(v), OneOf9::I(ref mut state)) => v.message(state, rest, message, app_state),
+            _ => unreachable!(),
+        }
+    }
+}
+
 // Because `OneOfState` is not public API, but is part of a public trait `impl`, it must be marked pub, but we don't want
 // to export it. Since this (`one_of`) module is public, we create a new module, allowing it to be pub but not exposed.
 #[doc(hidden)]
 mod hidden {
+    use crate::View;
+
+    use super::PhantomElementCtx;
+
     #[allow(unreachable_pub)]
     pub enum Never {}
+
+    impl<State, Action, Context: PhantomElementCtx, Message> View<State, Action, Context, Message>
+        for Never
+    {
+        type Element = Context::PhantomElement;
+
+        type ViewState = Never;
+
+        fn build(&self, _: &mut Context) -> (Self::Element, Self::ViewState) {
+            match *self {}
```

Comment:
That's a nice trick!

Reply:
Yeah - uninhabited types have some neat properties. We're proving that an impossible type has been constructed, and therefore that this code path was never entered.